bindings/netframework/ElectionGuard.NetFramework/ElectionGuardInterop/ManagedInstance.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+using namespace System;
+
+namespace ElectionGuardInterop
+{
+    template <class T>
+    public ref class ManagedInstance
+    {
+    public:
+        ManagedInstance() {}
+        ManagedInstance(T *instance, bool owned) : _instance(instance), _owned(owned) {}
```

Comment:
basically just the pointer to implementation pattern (pimpl) except you can toggle ownership on or off.  trying to hide that as much as i can from the external interface.